### DISCUSSION TITLE
About Signal

### DISCUSSION THREAD

**Discussion by xiyuzhai** [sycamore-rs/sycamore#437] [Ideas]
Signal can be exposed to function like store in Svelte. This is handy for things like receiving data from websockets, coordination, etc.

NodeRef should be made a signal in situations like canvas drawing so that use can use effects to draw.

I had to make these modifications on sycamore (just a couple of lines actually) to make my sycamore gui project work. (I transitioned from Svelte)

I'm a newbee. These are just my humble opinions.

**Comment by lukechu10** [comment]
Sorry for taking so long to reply. I think it would be a good idea to internally make `NodeRef` reactive by backing it up with a `Signal`. I'll play around a bit with this design and see what happens.
